docs/src/views/CustomComponents.tsx

```diff
@@ -0,0 +1,234 @@
+    <p>
+      Let's go step by step throughout all bits from the <code>createComponent</code> method.
+    </p>
+
+    <p>
+      The first argument to the <code>createComponent</code> config's param is the displayName. This
+      name is very important, because that's the key that will be used in the Provider's theme if
+      you want to define some styles or variables for this custom component. Let's see one example:
+    </p>
+
+    <ExampleSnippet
+      value={[
+        `<Provider`,
+        `  theme={{`,
+        `    componentStyles: {`,
+        `      StyledButton: {`,
+        `        root: ({ props, variables, theme: {siteVariables} }) => ({`,
+        `          backgroundColor: siteVariables.brand,`,
+        `          color: (variables as any).color,`,
+        `        }),`,
+        `      },`,
+        `    },`,
+        `    componentVariables: {`,
+        `      StyledButton: {`,
+        `        color: '#F2F2F2',`,
+        `      },`,
+        `    },`,
+        `  }}`,
+        `>`,
+        `  <StyledButton> Provider styled button </StyledButton>`,
+        '</Provider>',
+      ].join('\n')}
+      render={() => (
+        <Provider
+          theme={{
+            componentStyles: {
+              StyledButton: {
+                root: ({ props, variables, theme: { siteVariables } }) => ({
+                  backgroundColor: siteVariables.brand,
+                  color: (variables as any).color,
+                }),
+              },
+            },
+            componentVariables: {
+              StyledButton: {
+                color: '#F2F2F2',
+              },
+            },
+          }}
+        >
+          <StyledButton>Provider styled button</StyledButton>
+        </Provider>
+      )}
+    />
+
+    <p>
+      The second argument of the <code>createComponent</code> config param is the render method.
+      This is the place where you link the Stardust bits with your custom component. This method is
+      invoked with the following parameters:
+    </p>
+    <ul>
+      <li>
+        <code>stardust</code> - this is the object containing the evaluated theming props (classes
+        and rtl)
+      </li>
+      <li>
+        <code>...props</code> - all other props provided by the user
+      </li>
+    </ul>
+    <p>
+      The previous implementation of the render method inside the <code>createComponent</code>{' '}
+      function for the
+      <code>StyledButton</code>, allows the user, to use the styles and variables properties on the
+      custom component in the same way they would be used for any Stardust component.
+    </p>
+
+    <Header as="h2" content="Using the custom components" />
+
+    <p>
+      We already saw how the <code>Provider</code> can define some stylings and variables for the
+      custom components. Next, we will take a look into few examples of how the user can further
+      customize the styling and variables of the custom components, the same way they would do with
```

Comment:
nit: maybe simplified:
customize the styling and variables -> customize styles and variables